docs/developers_guide/ci/argo_workflows.md

```diff
@@ -1,37 +1,64 @@
-- **registry_put** places successfully builded image into the Harbor image repository
-- **crane_img_tag** tags images for unchanged services in the same *apps* structure
-- **version_change** promotes version in the version.yaml file in workload's gitops-repo.
-
-## Cluster workflow templates
-
-For every action workflows use cluster workflow templates consist of one or several steps to perform.
-In case of a list of parameters to process templates from clusterworkflow templates can be called iteratively.
-Cluster workflow templates are used to implement atomic phases of services images build process,
-except  `build_chain_p_cwft`, which have a DAG inside and called in a special way: inside so-called
-workflow-of-workflows structure in `.argo/build-wow-wf.yaml`.
-
-- build_chain_p_cwft
-    - megalinter-cwft — lints service code using Megalinter
-    - trivy-fs-s3-cwft — checks service fs with Trivy
-    - kaniko-s3-p-cwft — builds tar-image with Kaniko
-- crane-s3-p-cwft — tags and puts image into the Harbor image repo
-- crane-img-tag-cwft — tags unchanged (if any) services latest images and tag it with the same tag as changed
-- version-changer-cwft — change version in version.yaml in gitops-repo
-     
+## How It Works
+
+1. **GitHub Actions Job**: When triggered, a step from your GitHub Actions job submits an Argo Workflow.
+
+2. **Argo Workflow Submission**: The Argo Workflow acts as orchestrator, submitting tasks to CWFTs.
+
+3. **Cluster Workflow Templates (CWFTs)**  
+   Cluster Workflow Templates (CWFTs) are reusable building blocks that execute individual tasks, such as building images or scanning code. They are centrally defined and shared across workloads.
+
+4. **DAG Steps**  
+   Some CWFTs, like _build_chain_p_cwft_, use a Directed Acyclic Graph (DAG) to handle tasks with dependencies, allowing for parallel execution wherever possible.
+
+## Basic Workflows
+
+The `.argo` directory in your workload repository contains workflows for common CI/CD tasks:
+
+- **build-wow-wf**  
+   Handles building, linting, and running quality checks to prepare your workload for deployment.
+
+- **crane-p-wf**  
+   Pushes successfully built Docker images to the Harbor container registry.
+
+- **crane-img-tag-wf**  
+   Updates tags for unchanged services to keep their versions aligned with the new ones.
+
+- **version_changer-wf**  
+   Updates the `version.yaml` file in the GitOps repository with the latest versions for deployment.
+
+## Key Cluster Workflow Templates
+
+- **build_chain_p_cwft**  
+   Orchestrates the build process through a series of steps:
+   - **megalinter-cwft**: Lints the code to ensure it meets quality standards.
+   - **trivy-fs-s3-cwft**: Scans the filesystem for vulnerabilities.
+   - **kaniko-s3-p-cwft**: Builds container tar-images with Kaniko and puts them to the S3.
+
+- **crane-s3-p-cwft**  
+   Tags and pushes built images from S3 to the Harbor container registry.
```

Comment:
AFAIR this step reads tar archive from blob storage (S3 in AWS case), builds image from it, and pushes it to the registry (Harbor). So this description is not 100% true. It's better to check with @vtsarfin 

Also, please avoid using cloud provider specific service names, like S3 as it will be different on Azure and GCP.